uploadNeuroDB/NeuroDB/MRIProcessingUtility.pm

```diff
@@ -1742,68 +1742,65 @@ sub validateCandidate {
-routine of the C<$profile> file.
+Computes the SNR on the modalities specified in the Config module under the
+section Imaging Pipeline in the field called 'compute_snr_modalities'.
 
 INPUTS:
   - $tarchiveID: DICOM archive ID
   - $upload_id : upload ID of the study
-  - $profile   : configuration file (usually prod)
 
 =cut
 
 sub computeSNR {
 
     my $this = shift;
-    my ($row, $filename, $fileID, $base, $fullpath, $cmd, $message, $SNR, $SNR_old);
-    my ($tarchiveID, $upload_id, $profile)= @_;
-    my $data_dir = NeuroDB::DBI::getConfigSetting(
-                        $this->{dbhr},'dataDirBasepath'
-                        );
+    my ($tarchiveID, $upload_id) = @_;
 
-    my $query = "SELECT FileID, file from files f ";
-    my $where = "WHERE f.TarchiveSource=?";
-    $query = $query . $where;
-
-    if ($this->{debug}) {
-        print $query . "\n";
-    }
+    my $data_dir   = NeuroDB::DBI::getConfigSetting($this->{dbhr},'dataDirBasepath');
+    my $modalities = NeuroDB::DBI::getConfigSetting(
+        $this->{dbhr}, 'compute_snr_modalities'
+    );
 
+    (my $query = <<QUERY) =~ s/\n//gm;
+  SELECT    FileID, File, Scan_type
+  FROM      files f
+  JOIN      mri_scan_type mst ON (mst.ID=f.AcquisitionProtocolID)
+  WHERE     f.TarchiveSource=?
+QUERY
+    print $query . "\n" if ($this->{debug});
     my $minc_file_arr = ${$this->{'dbhr'}}->prepare($query);
     $minc_file_arr->execute($tarchiveID);
 
-    while ($row = $minc_file_arr->fetchrow_hashref()) {
-        $filename = $row->{'file'};
-        $fileID = $row->{'FileID'};
-        $base = basename($filename);
-        $fullpath = $data_dir . "/" . $filename;
-        if (defined(&Settings::getSNRModalities)
-            && Settings::getSNRModalities($base)) {
-                $cmd = "noise_estimate --snr $fullpath";
-                $SNR = `$cmd`;
-                $SNR =~ s/\n//g;
-                print "$cmd \n" if ($this->{verbose});
-                print "SNR is: $SNR \n" if ($this->{verbose});
-                my $file = NeuroDB::File->new($this->{dbhr});
-                $file->loadFile($fileID);
-                $SNR_old = $file->getParameter('SNR');
-                if ($SNR ne '') {
-                    if (($SNR_old ne '') && ($SNR_old ne $SNR)) {
-                        $message = "The SNR value will be updated from " .
-                            "$SNR_old to $SNR. \n";
-                        $this->{LOG}->print($message);
-                        $this->spool($message, 'N', $upload_id, $notify_detailed);
-                    }
-                    $file->setParameter('SNR', $SNR);
+    while (my $row = $minc_file_arr->fetchrow_hashref()) {
+        my $filename     = $row->{'File'};
+        my $fileID       = $row->{'FileID'};
+        my $fileScanType = $row->{'Scan_type'};
+        my $base         = basename($filename);
+        my $fullpath     = "$data_dir/$filename";
+        my $message;
+        if ( grep(/$fileScanType/, @{$modalities}) ) {
```

Comment:
Why not check for equality instead of a pattern match, as in 

```perl
if( grep($_ eq $fileScanType, @$modalities) ) {
```

If you have modalities like `T1` and `T1weighted` and you are checking for the modalities that match pattern `/T1/` then both will match and that's probably not what you want...